src/go/Rumble/rumble_server.go

```diff
@@ -0,0 +1,394 @@
+package main
+
+// rumble_server - Listen JSONiq requests and execute Rumble in spark cluster
+//
+// Copyright (c) 2021 - Ceyhun Uzunoglu <ceyhunuzngl@gmail.com>
+//
+
+import (
+    "bytes"
+    "encoding/json"
+    "errors"
+    "fmt"
+    "github.com/gofrs/flock"
+    "github.com/gorilla/mux"
+    "io"
+    "io/ioutil"
+    "log"
+    "net/http"
+    "os"
+    "os/exec"
+    "strings"
+    "time"
+)
+
+// Global variables
+var portsInUseLockFname string
+
+type ErrorMessage struct {
```

Comment:
For all struct please add `validate:xxx` parts, see
https://github.com/go-playground/validator and https://github.com/go-playground/validator/blob/master/_examples/simple/main.go
This is a very nice approach to automatically validate all attributes and will protect code from malformed input.